File: csharp/TDD.Tests/DiscountCalculatorTest.cs
using FluentAssertions;
using Xunit;

public class DiscountCalculatorTest
{
    [Fact]
    public void Discount_Should_Be_10_Percent_Of_Order_For_VIP_Customers()
{
        var discount = DiscountCalculator.CalculateDiscount(100, "VIP");
        discount.Should().Be(10d);
    }

    [Fact]
    public void Discount_Should_Be_0_For_Standard_Customer_When_Order_Under_100()
{
        var discount = DiscountCalculator.CalculateDiscount(99, "STANDARD");
        discount.Should().Be(0d);
    }

    [Fact]
    public void Discount_Should_Be_5_For_Standard_Customer_When_Order_Is_100()
{
        var discount = DiscountCalculator.CalculateDiscount(100, "STANDARD");
        discount.Should().Be(5d);
    }
}

File: java/src/test/java/DiscountCalculatorTest.java
import static org.hamcrest.Matchers.is;

import org.junit.jupiter.api.Test;

import static org.hamcrest.MatcherAssert.assertThat;

class DiscountCalculatorTest {
    
    @Test
    void discount_should_be_10_percent_of_order_for_VIP_customers(){
        double discount = DiscountCalculator.calculateDiscount(100, "VIP");
        assertThat(discount, is(10d));
    }

    @Test
    void discount_should_be_0_for_standard_customer_when_order_under_100(){
        double discount = DiscountCalculator.calculateDiscount(99, "STANDARD");
        assertThat(discount, is(0d));
    }

    @Test
    void discount_should_be_5_for_standard_customer_when_order_is_100(){
        double discount = DiscountCalculator.calculateDiscount(100, "STANDARD");
        assertThat(discount, is(5d));
    }
}
